talk/objectorientation/advancedoo.tex

```diff
@@ -305,6 +306,54 @@
   \end{multicols}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Polymorphism and destruction}
+  \begin{block}{Owning base pointers}
+    \begin{itemize}
+    \item We sometimes need to maintain owning pointers to base classes:
+    \begin{cppcode*}{gobble=2}
+      struct Drawable {
+         virtual void draw() = 0;
+      };
+
+      Drawable* p = getImpl(); // returns derived obj.
+      p->draw();
+      delete p;
```

Comment:
Maybe put all the code into an `exampleblock`? I don't really insist, but I think most of the time we have the blue boxes for text and another block for code.

Reply:
I should at least put it out of the block of the bullet point list.

Reply:
Better?

Reply:
Yes!